### PR TITLE
Add categories to fields section

### PR DESCRIPTION
**Pull request by Hegazy360**
screenshot-localhost_4200-2019 09 25-17_38_33
screenshot-localhost_4200-2019 09 25-17_38_53
screenshot-localhost_4200-2019 09 25-17_39_12

### CODE REVIEW

File: addon/components/hyper-table.js [hunk 1]
```diff
@@ -45,13 +47,16 @@ export default Component.extend({
...
+    '_activeGroup',
     function() {
       let columns = A(this._columns);
 
-      if (!isEmpty(this._availableColumnsKeyword)) {
-        let reg = RegExp(this._availableColumnsKeyword, 'i');
-        columns = A(columns.filter((x) => reg.test(x.title)));
-      }
+      columns = A(columns.filter((x) => {
+        const hasKeyword = x.title.toLowerCase().indexOf(this._availableColumnsKeyword.toLowerCase()) == 0;
```
Comment (phndiaye) [inline addon/components/hyper-table.js]: use `===`

File: tests/dummy/app/services/plans-fetcher.js [hunk 2]
```diff
@@ -45,64 +45,87 @@ const MOCK_DATA = A([
...
+  }
+]
+
 const DEFAULT_COLUMNS = [
   {
     title: 'Plan Name',
     property: 'name',
-    type: 'text'
+    type: 'text',
+    group: 'group_1'
```
Comment (phndiaye) [inline tests/dummy/app/services/plans-fetcher.js]: `groups: []`

File: tests/dummy/app/services/plans-fetcher.js [hunk 3]
```diff
@@ -45,64 +45,87 @@ const MOCK_DATA = A([
   })
 ]);
 
+const DEFAULT_GROUPS = [
+  {
+    type: 'group_1',
```
Comment (phndiaye) [inline tests/dummy/app/services/plans-fetcher.js]: `group.key`

File: addon/types/table.js [hunk 4]
```diff
@@ -62,5 +63,9 @@ export default EmberObject.extend({
   updateOrdering(column, orderBy) {
     this.columns.forEach((c) => c.set('orderBy', null));
     column.set('orderBy', orderBy);
+  },
+
+  updateGroups(groups){
```
Comment (phndiaye) [inline addon/types/table.js]: `updateColumnCategories`

File: addon/types/table.js [hunk 5]
```diff
@@ -7,6 +7,7 @@ import Column from '@upfluence/hypertable/types/column';
 export default EmberObject.extend({
   columns: [],
   data: [],
+  groups: [],
```
Comment (phndiaye) [inline addon/types/table.js]: `columnCategories`

File: addon/components/hyper-table.js [hunk 6]
```diff
@@ -152,6 +157,10 @@ export default Component.extend({
 
     openAvailableColumns() {
       this.toggleProperty('_availableColumnsPanel');
+    },
+
+    activateGroup(group) {
```
Comment (phndiaye) [inline addon/components/hyper-table.js]: `setColumnCategory`

File: addon/components/hyper-table.js [hunk 7]
```diff
@@ -45,13 +47,16 @@ export default Component.extend({
...
     function() {
       let columns = A(this._columns);
 
-      if (!isEmpty(this._availableColumnsKeyword)) {
-        let reg = RegExp(this._availableColumnsKeyword, 'i');
-        columns = A(columns.filter((x) => reg.test(x.title)));
-      }
+      columns = A(columns.filter((x) => {
+        const hasKeyword = x.title.toLowerCase().indexOf(this._availableColumnsKeyword.toLowerCase()) >= 0;
+        const hasActiveGroup = x.categories.indexOf(this._activeColumnCategory) >= 0 || !this._activeColumnCategory;
```
Comment (tgallice) [inline addon/components/hyper-table.js]: maybe the inverse `!this._activeColumnCategory || x.categories.indexOf(this._activeColumnCategory) >= 0` the check if there is an activeColumn is less expensive

File: addon/components/hyper-table.js [hunk 8]
```diff
@@ -45,13 +47,16 @@ export default Component.extend({
...
+    '_activeColumnCategory',
     function() {
       let columns = A(this._columns);
 
-      if (!isEmpty(this._availableColumnsKeyword)) {
-        let reg = RegExp(this._availableColumnsKeyword, 'i');
-        columns = A(columns.filter((x) => reg.test(x.title)));
-      }
+      columns = A(columns.filter((x) => {
+        const hasKeyword = x.title.toLowerCase().indexOf(this._availableColumnsKeyword.toLowerCase()) >= 0;
```
Comment (tgallice) [inline addon/components/hyper-table.js]: `_availableColumnsKeyword` can be empty like the activeColumn, no ? Maybe do the same

File: app/templates/components/hyper-table.hbs [hunk 9]
```diff
@@ -37,12 +43,19 @@
...
-              <div class="field">
-                {{upf-checkbox value=column.visible size="sm"}}
-                <div class="margin-left-xxx-sm">{{column.title}}</div>
+            {{#if _orderedFilteredColumns}}
+              {{#each _orderedFilteredColumns as |column|}}
+                <div class="field">
+                  {{upf-checkbox value=column.visible size="sm"}}
+                  <div class="margin-left-xxx-sm">{{column.title}}</div>
+                </div>
+              {{/each}}
```
Comment (tgallice) [inline app/templates/components/hyper-table.hbs]: This shsould work 

```
{{#each}}
{{else}}
{{/each}}
```

**Pull request merged**